src/Ship.js

```diff
@@ -0,0 +1,30 @@
+class Ship {
+  constructor(itinerary){
+    this.itinerary = itinerary;
+    this.currentPort = itinerary.ports[0];
+    this.previousPort = null;
+    this.currentPort.addShip(this);
+  };
+  setSail() {
+    const itinerary = this.itinerary;
+    const currentPortIndex = itinerary.ports.indexOf(this.currentPort);
+
+    if (currentPortIndex === (itinerary.ports.length -1)){
+      throw new Error('End of itinterary reached');
+    };
+
+    this.previousPort = this.currentPort;
+    this.currentPort.removeShip(this);
+    this.currentPort = null;
+  };
+
+  dock() {
+    const itinerary = this.itinerary;
+    const previousPortIndex = itinerary.ports.indexOf(this.previousPort);
+
+    this.currentPort = itinerary.ports[previousPortIndex + 1];
+    this.currentPort.addShip(this);
```

Comment:
As with previous comment:
```suggestion
    const previousPortIndex = this.itinerary.ports.indexOf(this.previousPort);

    this.currentPort = this.itinerary.ports[previousPortIndex + 1];
    this.currentPort.addShip(this);
```